lib/OrdinaryDiffEqDifferentiation/src/derivative_utils.jl

```diff
@@ -172,7 +172,21 @@ function calc_J!(J, integrator, cache, next_step::Bool = false)
         if DiffEqBase.has_jac(f)
             duprev = integrator.duprev
             uf = cache.uf
-            f.jac(J, duprev, uprev, p, uf.α * uf.invγdt, t)
+            # need to do some jank here to account for sparsity pattern of W
+            # https://github.com/SciML/OrdinaryDiffEq.jl/issues/2653
+
+            # we need to set all nzval to a non-zero number
+            # otherwise in the following line any zero gets interpreted as a structural zero
+            if !isnothing(integrator.f.jac_prototype) && 
+                integrator.f.jac_prototype isa SparseMatrixCSC
+
+                integrator.f.jac_prototype.nzval .= 1.0
+                J .= 1.0 .* integrator.f.jac_prototype
+                J.nzval .= 0.0
```

Comment:
```suggestion
                integrator.f.jac_prototype.nzval .= true
                J .= true .* integrator.f.jac_prototype
                J.nzval .= false
```
These are "minimally-typed" zeros and ones